src/main/scala/com/raphtory/core/model/communication/RaphtoryMessages.scala

```diff
@@ -0,0 +1,131 @@
+package com.raphtory.core.model.communication
+
+import com.raphtory.api.Analyser
+import com.raphtory.core.actors.PartitionManager.Workers.ViewJob
+import com.raphtory.core.model.graphentities.Edge
+
+import scala.collection.mutable
+
+/**
+  * Created by Mirate on 30/05/2017.
+  */
+sealed trait Property {
+  def key: String
+  def value: Any
+}
+
+case class Type(name: String)
+case class ImmutableProperty(key: String, value: String) extends Property
+case class StringProperty(key: String, value: String)    extends Property
+case class LongProperty(key: String, value: Long)        extends Property
+case class DoubleProperty(key: String, value: Double)    extends Property
+case class Properties(property: Property*)
+
+sealed trait GraphUpdate {
+  val updateTime: Long
+  val srcId: Long
+}
+
+case class VertexAdd(updateTime: Long, srcId: Long, properties: Properties, vType: Option[Type]) extends GraphUpdate //add a vertex (or add/update a property to an existing vertex)
+case class VertexDelete(updateTime: Long, srcId: Long) extends GraphUpdate
+case class EdgeAdd(updateTime: Long, srcId: Long, dstId: Long, properties: Properties, eType: Option[Type]) extends GraphUpdate
+case class EdgeDelete(updateTime: Long, srcId: Long, dstId: Long) extends GraphUpdate
+
+case class TrackedGraphUpdate[+T <: GraphUpdate](channelId: String, channelTime:Int, update: T)
```

Comment:
Much of this has now been changed with the new reworks by @wuliaososhunhun. The short answer is because it probably didn't work without it, but is now gone.